src/main/java/uk/gov/companieshouse/repository/AppealRepository.java

```diff
@@ -11,5 +12,9 @@
 
     Appeal insert(Appeal appeal);
     Optional<Appeal> findById(String id);
+
+    @Query("{ 'penaltyIdentifier' : {'companyNumber' : ?0, 'penaltyReference' : ?1 } }")
+    Optional<Appeal> findByPenaltyReference(String companyNumber, String penaltyReference);
```

Comment:
Is the companyNumber needed for the query?

Reply:
I don't think it will be needed as there is a 1-to-1 relationship between the penalty reference and an appeal